src/lib/scopes.ts

```diff
@@ -46,24 +47,15 @@ export interface ExternalScope {
  */
 export function convertExternalToScope (
   externalScopes: ExternalScope[], consentId: string): Scope[] {
-  const scopes: Scope[] = []
-
-  const scopes: Scope[] = [].concat(externalScopes.map(
-     (element: ExternalScope): Scopes[] => element.actions.map((action: string): Scope => ({
-            consentId,
-            accountId: element.accountId,
-            action
-     })
-  ))
-    const accountId = element.accountId
-    element.actions.forEach((action: string): void => {
-      const scope = {
+  const scopes: Scope[] = ([] as Scope[]).concat(...externalScopes.map(
+    (element: ExternalScope): Scope[] =>
+      element.actions.map((action: string): Scope => ({
         consentId,
-        accountId,
+        accountId: element.accountId,
         action
-      }
-      scopes.push(scope)
-    })
-  })
+      })
+      )
+  ))
```

Comment:
```suggestion
  ).flat()
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat